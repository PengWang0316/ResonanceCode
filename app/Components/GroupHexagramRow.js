import React from 'react';
import PropTypes from 'prop-types';

import HexagramImage from './HexagramImage';

const GroupHexagramRow = ({ hexagram, handleHexagramClick, hexagramsImgArrMap }) => {
  const isImgArrAvailable = Object.keys(hexagramsImgArrMap).length !== 0;
  return (
    <tr number={isImgArrAvailable ? hexagramsImgArrMap[hexagram.imageArr].number : ''} onClick={handleHexagramClick}>
      <td>{hexagram.name}</td>
      <td>#{isImgArrAvailable ? hexagramsImgArrMap[hexagram.imageArr].number : ''}</td>
      <td>
        <div style={{ minWidth: '35px' }}>
          <HexagramImage
            isBlack
            imageNumber={hexagram.imageArr}
            isFirstImage
            isSmall
          />
        </div>
      </td>
      <td>{isImgArrAvailable ? hexagramsImgArrMap[hexagram.imageArr].resonance_code_name : ''}</td>
    </tr>
  );
};
GroupHexagramRow.propTypes = {
  hexagram: PropTypes.object.isRequired,
  handleHexagramClick: PropTypes.func.isRequired,
  hexagramsImgArrMap: PropTypes.object.isRequired
};
// GroupHexagramRow.defaultProps = { hexagramsImgArrMap: {} };
export default GroupHexagramRow;
