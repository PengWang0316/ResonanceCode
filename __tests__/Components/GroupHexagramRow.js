import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import GroupHexagramRow from '../../app/Components/GroupHexagramRow';

jest.mock('../../app/Components/HexagramImage', () => 'HexagramImage');

describe('GroupHexagramRow test', () => {
  const defaultProps = {
    hexagram: { number: 1, name: 'name', imageArr: '7,9-7,9' },
    handleHexagramClick: jest.fn()
  };
  // const getShallowComponent = (props = defaultProps) => shallow(<GroupHexagramRow {...props} />);

  test('GroupHexagramRow snapshot', () => expect(renderer.create(<GroupHexagramRow {...defaultProps} />).toJSON()).toMatchSnapshot());
});
