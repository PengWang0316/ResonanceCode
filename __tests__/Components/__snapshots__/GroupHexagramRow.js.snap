// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupHexagramRow test GroupHexagramRow snapshot 1`] = `
<tr
  number={1}
  onClick={[MockFunction]}
>
  <td>
    name
  </td>
  <td>
    #
    1
  </td>
  <td>
    <HexagramImage
      imageNumber="7,9-7,9"
      isBlack={true}
      isFirstImage={true}
      isSmall={true}
    />
  </td>
  <td>
    resonance_code_name
  </td>
</tr>
`;

exports[`GroupHexagramRow test GroupHexagramRow without hexagramsImgArrMap snapshot 1`] = `
<tr
  number=""
  onClick={[MockFunction]}
>
  <td>
    name
  </td>
  <td>
    #
    
  </td>
  <td>
    <HexagramImage
      imageNumber="7,9-7,9"
      isBlack={true}
      isFirstImage={true}
      isSmall={true}
    />
  </td>
  <td>
    
  </td>
</tr>
`;
